### DISCUSSION TITLE
Can I connect RabbitMQ with multiple URIs, for example, if I have three copies of RabbitMQ, can I fill in three IPs in the URL at once?

### DISCUSSION THREAD

**Discussion by charlienss** [rabbitmq/amqp091-go#218] [General]
Can I connect RabbitMQ with multiple URIs, for example, if I have three copies of RabbitMQ, can I fill in three IPs in the URL at once?

**Comment by Zerpet** [comment]
It is not possible to use multiple IPs/URLs at once in a single `Dial()` call. This client aims to be simple and provide the functions to interact with RabbitMQ. All the smart stuff, like connecting to multiple nodes, has to be taken care off from the application using this library. 

There are community wrappers around this library that implement smart behaviour (connection pooling, reconnection, etc.). I'm aware of: 

- https://github.com/jxsl13/amqpx
- https://github.com/pmorelli92/bunnify
- https://github.com/wagslane/go-rabbitmq

To be clear, I don't endorse the usage of those libraries over amqp091-go. Those are community libraries I keep an eye on, and look very promising. You should make your own evaluation whether the usage of those libraries is acceptable for your use case.
